apps/demo-game/src/pages/play/cockpit.tsx

```diff
@@ -144,13 +128,12 @@ function Cockpit() {
       } else if (type == 'total') {
         val = value.totalAssets
       }
-      return {
-        ...acc,
-        [value.ix]: val,
-      }
+      acc[value.ix] = val
```

Comment:
This was a suggestion from coderabbit XD - it claimed to be an unnecessary copy afaik and it made sense for me as I did not consider the mutableness. Let's quickly discuss why you don't think it's a good idea to make the accumulator mutable.

Reply:
see https://stackoverflow.com/questions/63727586/is-mutating-accumulator-in-reduce-function-considered-bad-practice

-> it's not bad here

this is really interesting showing that spread is very inefficient on large arrays (which is kind of intuitive as it copies stuff)

https://gist.github.com/customcommander/97eb4b3f1600773db59406d39f3f9cd7

personally I just try to avoid mutating anything as keeping the habit has probably resulted in less bugs, as I try to use pure functions everywhere. but it is fine to mutate stuff from time to time in general (and for cases where there is a performance impact especially :)).

Reply:
and this seems even more of a deep dive: https://www.richsnapp.com/article/2019/06-09-reduce-spread-anti-pattern